doc/services/rtio/index.rst

```diff
@@ -0,0 +1,431 @@
+.. _rtio_api:
+
+Real Time I/O (RTIO)
```

Comment:
This needs an entry in doc/develop/api/overview.rst as well, likely marking the status as 'experimental'. Please feel free to dismiss my review once you address  this.